### PR TITLE
Include and display regressed test pass rates

### PR DESCRIPTION
**Pull request by dgoodwin**
- **Include pass rates in regressed tests API struct**
- **Show the change in pass rate in regressed tests modal**
- **Make regression certainty col sortable**
- **Return stats even for non-regressed tests in assessComponentStatus**

These changes are working towards tracking pass rates when we open/close regressions, but wanted to put up for review in isolation to make things easier.

### CODE REVIEW

File: sippy-ng/src/component_readiness/RegressedTestsModal.js [hunk 1]
```diff
@@ -122,9 +122,26 @@ export default function RegressedTestsModal(props) {
...
+      valueGetter: (params) => {
+        if (!params.row.sample_stats || !params.row.base_stats) {
+          return ''
+        }
+        return (
+          (params.row.sample_stats.success_rate * 100).toFixed(0) -
+          (params.row.base_stats.success_rate * 100).toFixed(0)
+        )
+      },
+      renderCell: (param) => <div className="pass-rate">{param.value}%</div>,
```
Comment (dgoodwin) [inline sippy-ng/src/component_readiness/RegressedTestsModal.js:144]: This is the second column I've added in shift week. It's getting pretty wide and we're probably the only ones who care. These don't have to be permanent but they're super helpful right now as we explore the thresholds things are going regressed/not. They can be removed in future, or placed behind an "advanced" checkbox.

File: sippy-ng/src/component_readiness/CompReadyTestReport.js [hunk 2]
```diff
@@ -232,7 +232,7 @@ export default function CompReadyTestReport(props) {
     }
   }
 
-  const [statusStr, assessmentIcon] = getStatusAndIcon(data.report_status)
+  const [statusStr, assessmentIcon] = getStatusAndIcon(data.status)
```
Comment (dgoodwin) [inline sippy-ng/src/component_readiness/CompReadyTestReport.js:235]: This is now consistently named with the other fields.

File: pkg/api/component_report.go [hunk 3]
```diff
@@ -1940,8 +1933,37 @@ func (c *componentReportGenerator) assessComponentStatus(requiredConfidence, sam
 		sampleTotal = adjustedSampleTotal
 	}
 
+	sampleFailure := sampleTotal - sampleSuccess - sampleFlake
+	// The adjusted total for ignored runs can push failure count into the negatives if there were
+	// more ignored runs than actual failures. (or no failures at all)
+	if sampleFailure < 0 {
```
[the remark this answers is not in the document]
Comment (dgoodwin) [reply pkg/api/component_report.go:1939]: Good idea, I'll dump a bunch of info if we trigger this case.

[the remark this answers is not in the document]
Comment (dgoodwin) [reply pkg/api/component_report.go:1939]: FWIW it seemed to be 100% passing tests that I saw it on, they were getting pass rates over 100%. Would that make sense for *other* tests that passed fine in a job run we ignored?

[the remark this answers is not in the document]
Comment (neisw) [reply pkg/api/component_report.go:1939]: We really shouldn't see a non 0 value for numberOfIgnoredSampleJobRuns for tests that don't have failures.  Unless there is some bleed over due to variants.  Also for 4.17 I don't see any triaged records so unless you are testing with older 4.16 data I don't understand why that value is getting adjusted.

[the remark this answers is not in the document]
Comment (dgoodwin) [reply pkg/api/component_report.go:1939]: Going to skip that logging now that we know it's all the time and https://issues.redhat.com/browse/TRT-1743 is filed, it'll get super spammy if we do it now.

**Comment by neisw** [comment]
/lgtm

/hold
release when ready

**Pull request merged**